flow_action_components/GetOpportunityPartners/force-app/main/default/classes/GetOpportunityPartnersActionTest.cls

```diff
@@ -0,0 +1,169 @@
+*       <runTest>GetOpportunityPartnersActionTest<runTest>
+* *************************************************
+* @History
+* -------
+* VERSION | AUTHOR                | DATE               | DESCRIPTION
+* 1.0     | Tamar Erlich          | October 07, 2020   | created
+**/
+@IsTest
+public with sharing class GetOpportunityPartnersActionTest {
+
+
+    @isTest
+    public static void opportunityPartnersFound(){
+
+        // initialize variables
+        List<GetOpportunityPartnersAction.Requests> requestList = new List<GetOpportunityPartnersAction.Requests>();
+        String accountId;
+        String opportunityId;
+
+        // create test account
+        List<Account> accounts = new List<Account>();
+
+        for (Integer j = 0; j < 1; j++) {
+
+            Account a = new Account(
+                Name = 'email' + j + '.com'
+                );
+
+            accounts.add(a);
+        }
+        
+        insert accounts;
+
+        // create test opportunity
+        Date closeDate = System.today();
+        Opportunity testOpp;
+
+        testOpp = new Opportunity(
+            Name = 'test opp',
+            StageName = 'Open In-Progress',
+            Amount = 100,
+            CloseDate = closeDate
+        );
+        insert testOpp;
+
+        accountId = accounts[0].Id;
+        opportunityId = testOpp.Id;
+
+        // create test opportunity partner
+        OpportunityPartner testPartner;
+        testPartner = new OpportunityPartner(
+            OpportunityId = opportunityId,
+            AccountToId = accountId,
+            Role = 'Dealer',
+            IsPrimary = false
+        );
+        insert testPartner;
+
+        // prepare request for GetOpportunityPartnersAction
+        GetOpportunityPartnersAction.Requests request = new GetOpportunityPartnersAction.Requests();
+        request.accountId = accountId;
+        request.opportunityId = opportunityId;
+        requestList.add(request);
+
+        // run test and assert results
+        List <GetOpportunityPartnersAction.Results> results = GetOpportunityPartnersAction.get(requestList);
+        System.assertEquals(results[0].errors, null,'Errors not expected');
+        System.assertNotEquals(results[0].successful, 'false', 'Success expected');
+        System.assertEquals(true, results[0].outputCollection.size()>0, 'Opportunity Partners expected');
+    }
+
+    @isTest
+    public static void opportunityPartnersNotFound(){
+
+
+        // initialize variables
+        List<GetOpportunityPartnersAction.Requests> requestList = new List<GetOpportunityPartnersAction.Requests>();
+        String accountId;
+        String opportunityId;
+
+        // create test account
+        List<Account> accounts = new List<Account>();
+
+        for (Integer j = 0; j < 1; j++) {
+
+            Account a = new Account(
+                Name = 'email' + j + '.com'
+                );
+
+            accounts.add(a);
+        }
+        
+        insert accounts;
```

Comment:
fyi we have run into some problems where people try to install these actions, but on their orgs, there are some additional fields on the object (here, Account) that have to exist in order to create a record. This causes their test to fail because it can't create an Account. 

The main solution for this is to distribute as a managed package, which doesn't run tests. There's overhead involved in doing that, though, because you need to create a separate dev edition org for each managed package. 


I don't have any suggestions here. Just letting you know it's an issue. It's a good thing to have tests.

Reply:
I have encountered this and had to rewrite the test class for datatable in my org to accommodate for required account field :)